docs/sphinx/using/hardware.rst

```diff
@@ -0,0 +1,96 @@
+CUDA Quantum Hardware Backends
```

Comment:
I know this is still in draft, but just remember to show the async functions and persisting the future handle to file, and loading a future from file to retrieve results.